File: src/CodeGeneration/DataFlow/Blocks/A_plus_B_mult_B.java
package CodeGeneration.DataFlow.Blocks;

import CodeGeneration.DataFlow.Network.Nodes.Blocks.AbstractBlock;
import CodeGeneration.DataFlow.Network.Nodes.SignalNodes.Channels.ListChannel;
import CodeGeneration.DataFlow.Network.Nodes.BlocksAndSignalNodes.Operations.BinaryOperations.MatrixOperations.Multiplication;
import CodeGeneration.DataFlow.Network.Nodes.BlocksAndSignalNodes.Operations.BinaryOperations.UnitWiseOperations._Addition;
import CodeGeneration.DataFlow.Network.Nodes.BlocksAndSignalNodes.Operations.AbstractOperation;

/// (X, Y) -> ApBmB -> Z
/// X - ApBmB.A ->

public class A_plus_B_mult_B extends AbstractBlock {
    public A_plus_B_mult_B() {
        // Channel
        this.addNewInputLabel("A", new ListChannel());
        this.addNewInputLabel("B", new ListChannel());
        this.addNewOutputLabel("out", new ListChannel());

        // Blueprint
        AbstractOperation add = new _Addition();
        AbstractOperation mult = new Multiplication();

        this.connectTo(add, "A", "in1");
        this.connectTo(add, "B", "in2");

        add.connectTo(mult, "out", "in1");

        this.connectTo(mult, "B", "in2");

        mult.connectTo(this, "out", "out");
    }
}

File: src/CodeGeneration/DataFlow/kotlin/AplusBmultB.kt
package CodeGeneration.DataFlow.kotlin

import CodeGeneration.DataFlow.Network.AbstractBlock
import CodeGeneration.DataFlow.Network.ChannelId
import CodeGeneration.DataFlow.Network.ListChannel
import CodeGeneration.DataFlow.Operations.Addition
import CodeGeneration.DataFlow.Operations.Multiplication
import CodeGeneration.DataFlow.Operations.Operation

/// (X, Y) -> ApBmB -> Z
/// X - ApBmB.A ->

class AplusBmultB : AbstractBlock() {
    init {
        // Channel
        this
                .addInput("A", ListChannel())
                .addInput("B", ListChannel())
                .addOutput("C", ListChannel())

        // Blueprint
        val add = Addition()

        this.connectAll(add)

        val mult = Multiplication()

        add.connectTo(mult, "C", "A")

        this.connectTo(mult, "B", "B")

        mult.connectTo(this, ChannelId.C, ChannelId.C)
    }
}

